### PR TITLE
Faster f16 rounding on x86

### PR DESCRIPTION
**Pull request by Keno**
It turns out that due to AVX, doing F16 conversion using comparisons
(which the compiler turns into predicated SIMD computations) is 5x faster
(on randn(Float32) - the performance of this is data dependent).
on Intel machines with AVX2 (likely even faster with AVX512), because
the memory lookup becomes more expensive compared to the arithmetic.
The same is not true of other architectures or when the function isn't
vectorized. E.g. on PowerPC, this seems to cause ~10% slowdown.
However, given these considerations, it's not clear we actually want to
do this. All CPUs with AVX2 also have F16C, i.e. have built-in instructions
that perform this conversion in 2 cycles, so we should probably just
turn that on instead.

Benchmark scripts: https://gist.github.com/Keno/0bf23c7d0c75cfdab84580039d54997e

### CODE REVIEW

File: base/float.jl [hunk 1]
```diff
@@ -137,21 +137,86 @@ function Float32(x::Int128)
...
+            e = i - 127
+            (base, sh) = compute_base_shift(e)
+            _basetable[i|0x000+1] = base
+            _basetable[i|0x100+1] = base | 0x8000
+            _shifttable[i|0x000+1] = sh
+            _shifttable[i|0x100+1] = sh
+        end
+        global const shifttable = (_shifttable...,)
+        global const basetable = (_basetable...,)
+        global l
```
Comment (JeffBezanson) [inline base/float.jl]: ?

Comment (Keno) [reply base/float.jl]: whoops

**Comment by StefanKarpinski** [comment]
I'm a bit unclear on why my review is requested. I have no special Float16 expertise 😬

**Comment by Keno** [comment]
You and Jeff wrote the current version of this code.

**Comment by StefanKarpinski** [comment]
Hah, I guess that's fair.

**Comment by StefanKarpinski** [comment]
I've been assigned to review this, but @keno, if you think this is good, go for it.

**Comment by vtjnash** [comment]
Seemed conflicted

**Pull request closed without merging**